### PR TITLE
OSL-148: Setting up Redis cache backend for ApolloServer and cache shareableListPublic query response

### PR DESCRIPTION
**Pull request by katerinachinnappan**
## Goal

Setting up the Redis cache backend for `ApolloServer`. The `pocket-utils` `ElasticacheRedis` is not recommended to use and it is preferable to directly use the [`KeyvAdapter` package](https://github.com/apollographql/apollo-utils/blob/main/packages/keyvAdapter/src/index.ts) from `@apollo/utils.keyvadapter`. 

For now, cached responses for `shareableListPublic` query are stored for 60 seconds.

## Todos

- [x] deploy to dev

## Tickets

- [https://getpocket.atlassian.net/browse/OSL-148](https://getpocket.atlassian.net/browse/OSL-148)

### CODE REVIEW

File: src/public/resolvers/queries/ShareableList.ts [hunk 1]
```diff
@@ -43,14 +43,20 @@ export async function getShareableList(
...
   const list = await dbGetShareableListPublic(db, externalId, slug);
 
   if (!list) {
     throw new NotFoundError(externalId);
   }
-
+  // dynamically providing caching controll in the resolver
+  // PUBLIC scope because response is accessible to public not a single user
+  info.cacheControl.setCacheHint({
+    maxAge: 60, // The maximum amount of time the field's cached value is valid, in seconds.
```
Comment (katerinachinnappan) [inline src/public/resolvers/queries/ShareableList.ts:57]: setting to 1 minute for now. should it be increased?

File: schema-shared.graphql [hunk 2]
```diff
@@ -49,103 +49,103 @@ enum ShareableListModerationStatus {
...
-    The timestamp of when this story was added to the list by its owner.
-    """
-    createdAt: ISOString!
-    """
-    The timestamp of when the story was last updated. Not used for the MVP.
-    """
-    updatedAt: ISOString!
+  """
+  A unique string identifier in UUID format.
+  """
```
Comment (nina-py) [inline schema-shared.graphql:61]: Looks like these are just formatting changes 👍

File: src/public/resolvers/queries/ShareableList.ts (hunk 1 above)
Comment (nina-py) [reply src/public/resolvers/queries/ShareableList.ts:57]: We could start with that and adjust the numbers later 👍

File: README.md [hunk 3]
```diff
@@ -171,3 +170,11 @@ In `src/snowplow/types.ts` do the following:
...
 
 The unit tests for Snowplow events are defined in `src/snowplow/events.spec.ts`. [https://sinonjs.org/](Sinon JS)is used for test spies and stubs.
+
+### Redis Cache
+
+The Redis container runs on `localhost:6379`. To connect to the local Redis server (works for `dev` and `prod` as well, just find the Redis cluster primary endpoints in AWS ElastiCache and make sure to use the appropriate `$(maws)` role), one of the options is to install the [redis-gui](https://github.com/ekvedaras/redis-gui/releases) and setup the connection to `localhost:6379`.
+
+The `shareableListPublic` query caches the responses and stores them for 60 seconds in the Redis db. Here is a screenshot of how it gets stored:
+
+![Architecture](redis-gui-screenshot.png)
```
Comment (nina-py) [inline README.md:180]: Not a blocker, but it would be great to see that screenshot placed outside of the repository root. Maybe in a `/docs/` folder?

**Review by nina-py — APPROVED** [review]
Great work! 🚀

File: README.md (hunk 3 above)
Comment (katerinachinnappan) [reply README.md:180]: Good point, will follow up on this in another PR !

**Pull request merged**